packages/core-data/src/resolvers.js

```diff
@@ -543,13 +543,17 @@ export const __experimentalGetCurrentGlobalStylesId =
 		const globalStylesURL =
 			activeThemes?.[ 0 ]?._links?.[ 'wp:user-global-styles' ]?.[ 0 ]
 				?.href;
-		if ( globalStylesURL ) {
-			const globalStylesObject = await apiFetch( {
-				url: globalStylesURL,
-			} );
-			dispatch.__experimentalReceiveCurrentGlobalStylesId(
-				globalStylesObject.id
-			);
+		if ( ! globalStylesURL ) {
+			return;
+		}
+
+		// Regex matches the ID at the end of a URL or immediately before
+		// the query string.
+		const matches = globalStylesURL.match( /\/(\d+)(?:\?|$)/ );
```

Comment:
We could use [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) to at least remove the query?

Reply:
I almost reached for `URL` but ended up with the more familiar regex. What's the rationale behind switching to `URL`?

I'm happy to make the change tomorrow, just looking at how I could make better choices in the future.

Reply:
It just feels more like the right tool for it. It's just a nit.

Reply:
Sounds good. I've updated this via https://github.com/WordPress/gutenberg/pull/63071/commits/3f0f7b54b5b2e334ec0f115955de02a5e5a92d22